### PR TITLE
cli: add argument to `generate-packet-data` cli to use selected encoding format

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #3974 

### Commit Message / Changelog Entry

```bash
imp: add argument to `generate-packet-data` cli to use selected encoding format
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by srdtrk** [comment]
Should we backport this to `v7.3.x` ?  Can be released in `v7.3.1`

**Review by srdtrk — APPROVED** [review]
See my comment about possibly backporting this

File: modules/apps/27-interchain-accounts/host/client/cli/tx.go [hunk 1]
```diff
@@ -21,13 +22,13 @@ const (
...
 func generatePacketDataCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "generate-packet-data [message]",
-		Short: "Generates protobuf encoded ICA packet data.",
-		Long: `generate-packet-data accepts a message string and serializes it using protobuf
-into packet data which is outputted to stdout. It can be used in conjunction with send-tx"
-which submits pre-built packet data containing messages to be executed on the host chain.
-`,
-		Example: fmt.Sprintf(`%s tx interchain-accounts host generate-packet-data '{
+		Use:   "generate-packet-data [encoding] [message]",
```
Comment (srdtrk) [inline modules/apps/27-interchain-accounts/host/client/cli/tx.go:25]: Should encoding be added as an optional flag for backwards compatibility? Like `--proto3json` or `--protobuf`

**Comment by crodriguezvega** [comment]
> Should we backport this to `v7.3.x` ? Can be released in `v7.3.1`

Yes, sounds good. We can do that.

File: modules/apps/27-interchain-accounts/host/client/cli/tx.go [hunk 2]
```diff
@@ -74,7 +77,16 @@ which submits pre-built packet data containing messages to be executed on the ho
 				return err
 			}
 
-			packetDataBytes, err := generatePacketData(cdc, []byte(args[0]), memo)
+			encoding, err := cmd.Flags().GetString(encodingFlag)
+			if err != nil {
+				return err
+			}
+
+			if !slices.Contains([]string{icatypes.EncodingProtobuf, icatypes.EncodingProto3JSON}, encoding) {
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/host/client/cli/tx.go:85]: `slices` is still in the exp pkg on go1.19, should we use the internal `collections.Contains()` that we have at the moment, and update all occurrences to use `slices.Contains` when eden is merged? 

I don't feel too strongly about this, I'm also happy to keep this as is 👍

File: modules/apps/27-interchain-accounts/host/client/cli/tx.go [hunk 3]
```diff
@@ -85,19 +97,20 @@ which submits pre-built packet data containing messages to be executed on the ho
 		},
 	}
 
-	cmd.Flags().String(memoFlag, "", "an optional memo to be included in the interchain account packet data")
+	cmd.Flags().String(memoFlag, "", "an optional memo to be included in the interchain accounts packet data")
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/host/client/cli/tx.go:100]: for consistency with below

```suggestion
	cmd.Flags().String(memoFlag, "", "optional memo to be included in the interchain accounts packet data")
```

**Review by damiannolan — APPROVED** [review]

File: modules/apps/27-interchain-accounts/host/client/cli/tx.go (hunk 2 above)
Comment (crodriguezvega) [reply modules/apps/27-interchain-accounts/host/client/cli/tx.go:85]: Yeap, I can change it.

**Pull request merged**